lucene/core/src/java/org/apache/lucene/index/DirectoryReader.java

```diff
@@ -104,6 +105,17 @@ public static DirectoryReader open(final IndexCommit commit) throws IOException
     return StandardDirectoryReader.open(commit.getDirectory(), commit);
   }
 
+  /**
+   * Expert: returns an IndexReader reading the index in the given {@link IndexCommit}.
+   *
+   * @param commit the commit point to open
+   * @param minIndexVersionCreated the minimum index version created to check before opening the index
```

Comment:
I like how `listCommits` calls it a "major" version which avoids some ambiguity, maybe do the same here?

Also could you improve javadocs to better describe how this version parameter affects the behavior of this method?